gammapy/spectrum/tests/test_models.py

```diff
@@ -1,6 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import pytest
 import numpy as np
+import naima
```

Comment:
We have to restructure the testing here, because `naima` is only and optional dependency. That means the import must be "delayed" i.e. within a class or function, but not at the top-level of the module. This also means the `NaimaModel` must be tested separately from the other models. 

My suggestion would be to implement a `TestNaimaModel` class like so:

```python3
@requires_dependency("naima")
class TestNaimaModel:

    def test_pion_decay_(self):
        import naima
        model = NaimaModel()
        assert_quantity_allclose()
        assert_quantity_allclose()

    def test_ic(self):
        import naima
        model = NaimaModel()
        assert_quantity_allclose()
        assert_quantity_allclose()


    def test_synchrotron(self):
        import naima
        model = NaimaModel()
        assert_quantity_allclose()
        assert_quantity_allclose()

```